src/section/shared_state.rs

```diff
@@ -264,6 +301,22 @@ impl SharedState {
         elders_info: Proven<EldersInfo>,
         section_key: Proven<bls::PublicKey>,
     ) {
+        // The section public key of the proof shall be trusted by us.
+        match self
+            .our_history
+            .check_trust(iter::once(&section_key.proof.public_key))
+        {
+            TrustStatus::Trusted => {}
+            result => {
+                trace!(
+                    "not updating our section {:?} - Old section key {:?} is not trusted.",
+                    result,
+                    section_key.proof.public_key
+                );
+                return;
+            }
+        }
+
```

Comment:
Again, just use `self.our_history.has_key(&section_key.proof.public_key)`. But also verify the signatures and also verify `elders_info`, not just `section_key`. It might be useful to add a `verify` method to `Proven`, to avoid duplication.

Reply:
> But also verify the signatures and also verify elders_info, not just section_key. It might be useful to add a verify method to Proven, to avoid duplication.

I thought such signature check has been carried out already when accumulation. Will have a check through.